### DISCUSSION TITLE
Try integrating with AutoFilterer

### DISCUSSION THREAD

**Discussion by ParsaGachkar** [Megabit/Blazorise#4081] [Q&A]
I'm using [AutoFilterer](https://github.com/enisn/AutoFilterer) in my backend! I was wondering how to integrate [BlazoriseDataGrid](https://blazorise.com/docs/extensions/datagrid/getting-started) to Auto Filterer?! Can't find anything about remote filtering api for BlazoriseDataGrid!!!

**Comment by stsrki** [comment]
I'm not sure how AutoFilterer works, But Blazorise DataGrid doesn't have any opinion on where or how you will connect to a third-party data source. With DataGrid API, we give you all information about the current state of the grid, along with the filters applied. You can see our example https://blazorise.com/docs/extensions/datagrid/binding-data. Look for **Large Data** section.

I guess you will somehow have to translate the fields in the `DataGridReadDataEventArgs` to AutoFilterer.

**Reply by ParsaGachkar** [reply]
thanks for fast feedback. I'm already using `ReadData` Property and it works fine for basic paging, page-size and sorting the problem is I cannot get my filter input data! maybe you can guide me how?! am I using the correct Method?! or I need to use another method for filtering? here is the thing I'm trying to achieve:

1. Get Pagination and Sorting Data (Paginations and Sorting works)
2. Get User's Filter Request Data (but how to get filter data????)
3. Send All above to my Already implemented REST-API and get result to update grid's data!

**Comment by stsrki** [comment]
Did you try to use `SearchValue` on the column? 

eg.

```cs
private async Task OnReadData( DataGridReadDataEventArgs<Employee> e )
{
   // this will contain the filter data for the column
    var searchValue  = e.Columns[0].SearchValue;
}
```
